nncf/quantization/algorithms/pipeline.py

```diff
@@ -118,6 +118,9 @@ def run_step(
             current_graph = NNCFGraphFactory.create(current_model)
         current_model = pipeline_step[-1].apply(current_model, current_graph, step_statistics)
 
+        for algorithm in pipeline_step[:-1]:
```

Comment:
I think we should remove statistic points for each algorithm in ` pipeline_step` after it has been applied.
```python
        for algorithm in pipeline_step[:-1]:
            current_model = algorithm.apply(current_model, current_graph, step_statistics)
            current_graph = NNCFGraphFactory.create(current_model)
            step_statistics.remove_statistic_points(algorithm._algorithm_key)

        current_model = pipeline_step[-1].apply(current_model, current_graph, step_statistics)
        step_statistics.remove_statistic_points(pipeline_step[-1]._algorithm_key)
```

Reply:
Sure, will send it via commit in a bit

Reply:
> I think we should remove statistic points for each algorithm in ` pipeline_step` after it has been applied.
> 
> ```python
>         for algorithm in pipeline_step[:-1]:
>             current_model = algorithm.apply(current_model, current_graph, step_statistics)
>             current_graph = NNCFGraphFactory.create(current_model)
>             step_statistics.remove_statistic_points(algorithm._algorithm_key)
> 
>         current_model = pipeline_step[-1].apply(current_model, current_graph, step_statistics)
>         step_statistics.remove_statistic_points(pipeline_step[-1]._algorithm_key)
> ```

@KodiaqQ @andrey-churkin I'm pretty sure this code is wrong
Don't they get deleted _before_ being utilised? There's an error because of this, I'll try fixing it and ask for another review